### ISSUE TITLE
[JENKINS-17763] Git plugin "clean workspace" blows away private maven repository

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#2276]
<p>When you select "Clean repository workspace", this executes the equivalent<br/>
of 'git clean -fdx' in $WORKSPACE</p>

<p>If you have 'local maven repository' selected, then there is a maven repo<br/>
in $WORKSPACE/.repository.</p>

<p>So this option cleans the whole repository workspace, which is not what is wanted.</p>

<p>I'm not sure what the most reasonable fix is, though.</p>

<p>Clearly 'git clean -fdx -e .repository' will probably work, but it's a bit... 'magic'</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=magnayn">magnayn</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-17763" target="_blank">Git plugin "clean workspace" blows away private maven repository</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ndeloof.png" title="ndeloof's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ndeloof">ndeloof</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: git-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2013-10-22T14:49:58+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When you select "Clean repository workspace", this executes the equivalent
of 'git clean -fdx' in $WORKSPACE</p>

<p>If you have 'local maven repository' selected, then there is a maven repo
in $WORKSPACE/.repository.</p>

<p>So this option cleans the whole repository workspace, which is not what is wanted.</p>

<p>I'm not sure what the most reasonable fix is, though.</p>

<p>Clearly 'git clean -fdx -e .repository' will probably work, but it's a bit... 'magic'</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-17763?focusedId=179558&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179558">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is the same as e.g. the subversion-plugin does and letting them behave in the same way is IMO a good thing.
If you don't want this you can either checkout the git repository not to the root of the Jenkins workspace or put the Maven repository somewhere else.</p>
</pre>
</details>

</ul>
</details>

<p>This is the same as e.g. the subversion-plugin does and letting them behave in the same way is IMO a good thing.<br/>
If you don't want this you can either checkout the git repository not to the root of the Jenkins workspace or put the Maven repository somewhere else.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mburger.png" title="mburger's avatar" /> mburger:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-17763?focusedId=180653&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-180653">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I stumbled upon the same problem. I would like to wipe all obsolete files, but would like to keep the private Maven repository.</p>

<p>For projects where I want to wipe those files too, I use the <a href="https://wiki.jenkins-ci.org/display/JENKINS/Workspace+Cleanup+Plugin" class="external-link" target="_blank" rel="nofollow noopener">Workspace Cleanup Plugin</a>.</p>
</pre>
</details>

</ul>
</details>

<p>I stumbled upon the same problem. I would like to wipe all obsolete files, but would like to keep the private Maven repository.</p>

<p>For projects where I want to wipe those files too, I use the <a href="https://wiki.jenkins-ci.org/display/JENKINS/Workspace+Cleanup+Plugin" class="external-link" target="_blank" rel="nofollow noopener">Workspace Cleanup Plugin</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>magnayn:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-17763?focusedId=180659&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-180659">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>There is a fix :
<a href="https://github.com/jenkinsci/jenkins/pull/808" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/808</a></p>

<p>Which is to store the maven files elsewhere, in parallel to the build. Works for us.</p>
</pre>
</details>

</ul>
</details>

<p>There is a fix :<br/>
<a href="https://github.com/jenkinsci/jenkins/pull/808" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/808</a></p>

<p>Which is to store the maven files elsewhere, in parallel to the build. Works for us.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ndeloof.png" title="ndeloof's avatar" /> ndeloof:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-17763?focusedId=187917&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-187917">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>"clean workspace" is clear about the intention : make the workspace look like a fresh clone. So this is a maven-plugin issue</p>
</pre>
</details>

</ul>
</details>

<p>"clean workspace" is clear about the intention : make the workspace look like a fresh clone. So this is a maven-plugin issue</p>
